### PR TITLE
MR: Remove deprecate AssertHelpers

### PR DESCRIPTION
**Pull request by liuxiaocs7**
- SubTask of https://github.com/apache/iceberg/issues/7094

Remove `AssertHelpers` in `iceberg-mr` module

### CODE REVIEW

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java [hunk 1]
```diff
@@ -210,28 +210,25 @@ public void testCreateDropTable() throws TException, IOException, InterruptedExc
...
       // Check if the table was really dropped even from the Catalog
-      AssertHelpers.assertThrows(
-          "should throw exception",
-          NoSuchTableException.class,
-          "Table does not exist",
-          () -> {
-            testTables.loadTable(identifier);
-          });
+      Assertions.assertThatThrownBy(
+              () -> {
```
Comment (nastra) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java:214]: nit: the { } aren't necessary here and further below

**Review by nastra — APPROVED** [review]
LGTM with some nits, thanks @liuxiaocs7

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java [hunk 2]
```diff
@@ -272,13 +269,12 @@ public void testCreateDropTableNonDefaultCatalog() throws TException, Interrupte
...
-        "Table does not exist",
-        () -> {
-          testTables.loadTable(identifier);
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              testTables.loadTable(identifier);
+            })
+        .isInstanceOf(NoSuchTableException.class)
+        .hasMessageStartingWith("Table does not exist");
```
[the remark this answers is not in the document]
Comment (liuxiaocs7) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java:277]: There are two cases:

```shell
Expecting actual:
  "Table does not exist: default.customers"
```

```shell
Expecting actual:
  "Table does not exist at location: /tmp/junit7233921062738785052/default/customers"
```

so i use `hasMessageStartingWith` here.

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java (hunk 1 above)
Comment (liuxiaocs7) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java]: Done!

**Review by nastra — APPROVED** [review]

**Comment by jackye1995** [comment]
Thanks for the work @liuxiaocs7 and thanks for the review @nastra !

**Pull request merged**